x/interchainstaking/keeper/delegation.go

```diff
@@ -10,17 +12,22 @@ import (
 func (k *Keeper) Delegate(ctx sdk.Context, zone types.RegisteredZone, account *types.ICAAccount) error {
 	var msgs []sdk.Msg
 
-	for _, asset := range account.Balance {
+	balance := account.Balance
+	// deterministically sort balance
+	sort.Slice(balance, func(i, j int) bool { return account.Balance[i].Denom > account.Balance[j].Denom })
```

Comment:
SliceStable to maintain input ordering

Reply:
we are sorting on denom (which must be unique), not amount here. Isn't Slice is sufficient?